### PR TITLE
make policy template names unique

### PR DESCRIPTION
**Pull request by jamestalton**
Signed-off-by: James Talton <jtalton@redhat.com>
https://github.com/stolostron/backlog/issues/22523

### CODE REVIEW

**Review by zlayne — APPROVED** [review]

**Review by KevinFCormier — COMMENTED** [review]
Please see my comments. The 2.4 and before way of doing this resulted in longer template names, but I think it is more robust and simpler to implement.

File: wizards/Policy/PolicyWizard.tsx [hunk 1]
```diff
@@ -261,24 +260,24 @@ export function PolicyWizardTemplates() {
...
                                                 return get(existingTemplate, 'objectDefinition.metadata.name') === newName
-                                            })
-                                        ) {
-                                            newName = name + '-' + (counter++).toString()
-                                        }
-                                        set(newPolicyTemplate, 'objectDefinition.metadata.name', newName)
+                                            })) ||
+                                        isExistingTemplateName(newName, props.policies)
+                                    ) {
+                                        newName = name + '-' + (counter++).toString()
```
[the remark this answers is not in the document]
Comment (KevinFCormier) [reply wizards/Policy/PolicyWizard.tsx:278]: Gus confirmed that they need to be unique globally, so I agree this is better than using the policy name.

**Review by KevinFCormier — APPROVED** [review]

File: wizards/Policy/PolicyWizard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jamestalton) [reply wizards/Policy/PolicyWizard.tsx:278]: when creating add adding, there is a chance they did not set a name yet... so number is safer

File: wizards/Policy/PolicyWizard.tsx [hunk 2]
```diff
@@ -138,7 +138,7 @@ export function PolicyWizard(props: {
             </Step>
             <Step label="Policy templates" id="templates">
                 <ItemSelector selectKey="kind" selectValue={PolicyKind}>
-                    <PolicyWizardTemplates />
+                    <PolicyWizardTemplates policies={props.policies} />
```
[the remark this answers is not in the document]
Comment (jamestalton) [reply wizards/Policy/PolicyWizard.tsx:141]: all policies

**Pull request merged**